packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioLibraryPublicationHandler.java

```diff
@@ -0,0 +1,264 @@
+                    });
+            });
+        // Find domainTypes that refer to domains in another library.
+        rSet.getResources().stream()
+            .flatMap(r -> StreamSupport.stream(Spliterators.spliteratorUnknownSize(r.getAllContents(), 0), false))
+            .forEach(eObject -> {
+                EObject root = this.getContainingLibraryCandidate(eObject).orElse(eObject);
+                eObject.eClass().getEAllAttributes().stream()
+                    .filter(eAttribute -> eAttribute.getName().equals("domainType"))
+                    .findFirst()
+                    .ifPresent(domainTypeAttribute -> {
+                        String domainType = (String) eObject.eGet(domainTypeAttribute);
+                        if (domainType != null) {
+                            String[] splittedDomainType = domainType.split("::");
+                            String domainName = splittedDomainType[0];
+                            domains.stream().filter(d -> d.getName().equals(domainName)).findFirst()
+                                .ifPresent(d -> result.computeIfAbsent(root, k -> new HashSet<>()).add(d));
+                        }
+
+                    });
+            });
+        return result;
+    }
+
+    private void reduceTransitiveDependencies(Map<EObject, Set<EObject>> dependencies) {
+        for (Map.Entry<EObject, Set<EObject>> entry : dependencies.entrySet()) {
+            entry.getValue().removeIf(dependency ->
+                entry.getValue().stream()
+                    .filter(e -> e != dependency)
+                    .flatMap(e -> dependencies.getOrDefault(e, Set.of()).stream())
+                    .anyMatch(e -> e == dependency));
+        }
+    }
+
+    private Optional<EObject> getContainingLibraryCandidate(EObject eObject) {
+        final Optional<EObject> result;
+        if (eObject instanceof RepresentationDescription || eObject instanceof Domain) {
+            result = Optional.of(eObject);
+        } else if (eObject == null) {
+            result = Optional.empty();
+        } else {
+            result = this.getContainingLibraryCandidate(eObject.eContainer());
+        }
+        return result;
+    }
+
+    private Optional<String> toJson(Resource resource) {
+        Optional<String> result = Optional.empty();
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            Map<String, Object> saveOptions = new HashMap<>();
+            saveOptions.put(JsonResource.OPTION_ENCODING, JsonResource.ENCODING_UTF_8);
+            saveOptions.put(JsonResource.OPTION_SCHEMA_LOCATION, Boolean.TRUE);
+            saveOptions.put(JsonResource.OPTION_ID_MANAGER, new EObjectRandomIDManager());
+
+            resource.save(outputStream, saveOptions);
+
+            result = Optional.of(outputStream.toString());
+        } catch (IOException exception) {
+            this.logger.warn(exception.getMessage(), exception);
+        }
+        return result;
```

Comment:
We don't want to re-invent the wheel, the need the `ResourceToDocumentService` etc.